awsx/ecr/image.ts

```diff
@@ -62,37 +62,52 @@ export function computeImageFromAsset(
       throw new Error("Invalid credentials");
     }
     return {
-      registry: ecrCredentials.proxyEndpoint,
+      address: ecrCredentials.proxyEndpoint,
       username: username,
       password: password,
     };
   });
 
+  let cacheFrom: docker.types.input.CacheFromArgs[] = [];
+  if (dockerInputs.cacheFrom !== undefined) {
+    cacheFrom = dockerInputs.cacheFrom.map((c) => {
+      return {
+        registry: {
+          ref: c,
+        },
+      };
+    });
+  }
+  // Use an inline cache by default.
+  if (cacheFrom.length === 0) {
+    cacheFrom.push({ registry: { ref: canonicalImageName } });
+  }
+
+  let context = ".";
+  if (dockerInputs.context !== undefined) {
+    context = dockerInputs.context;
+  }
+
   const dockerImageArgs: docker.ImageArgs = {
-    imageName: canonicalImageName,
-    build: {
-      args: dockerInputs.args,
-      builderVersion: dockerInputs.builderVersion,
-      cacheFrom: dockerInputs.cacheFrom
-        ? {
-            images: dockerInputs.cacheFrom,
-          }
-        : undefined,
-      context: dockerInputs.context,
-      dockerfile: dockerInputs.dockerfile,
-      platform: dockerInputs.platform,
-      target: dockerInputs.target,
-    },
-    registry: registryCredentials,
+    tags: [canonicalImageName],
+    buildArgs: dockerInputs.args,
+    cacheFrom: cacheFrom,
+    cacheTo: [{ inline: {} }],
```

Comment:
Great question! It's actually full cache artifacts, so it can increase the size a bit. I think it would be best to take this out and stay as close to the existing behavior of not doing any caching stuff unless `cacheFrom` is set.

However this makes me realize we probably _do_ want inline caching when `cacheFrom` is set. Currently it's using a registry cache output, but this isn't compatible with the existing behavior because users currently point this at an image (not a cache artifact) due to the way the old Docker provider did caching. (It would simply pull images to get layers into your local build cache.) In other words we don't want to export a cache to `image:latest`, but it would be reasonable to push an image with inline caching.

Caching Docker images is hard enough as it is and the old provider's behavior was also easy to get wrong. For example `cacheFrom` might not do anything without a magic env var:
```ts
new docker.Image("my-app-image", {
    build: {
        args: {
            BUILDKIT_INLINE_CACHE: "1",
        },
        cacheFrom: {
            images: ["foo:latest"],
        },
        context: "app/",
        dockerfile: "app/Dockerfile",
    },
    imageName: "foo:latest",
});
```
So for us to stay as close to the existing behavior while also allowing caching to _somewhat_ work, I think we basically just want to enable this `BUILDKIT_INLINE_CACHE` behavior when `cacheFrom` is present.